docs/features/functions.md

```diff
@@ -0,0 +1,20 @@
+# Functions
+
+```js
+// Declare a function that takes in 2 arguments and returns the sum of them
+let test = [a:int b:int] -> int {
+	return a + b
+} // The type of this is int -> int -> int
+
+
+let test2 = test(1, 2)
+
+print(test)
+print(test2)
+
+print(type(test)) // This print out `int -> int -> int`
+```
+
+## Notes:
+- The type of a function is `[type of argument 1] -> [type of argument 2] ... -> [return type]`
```

Comment:
Could say that this is due to currying since calling a function with one argument will return `[type of argument 2] -> ... -> [return type]`

Reply:
Added.